plugin/src/main/groovy/com/btkelly/gnag/api/GitHubApi.java

```diff
@@ -53,41 +57,78 @@ public GitHubApi(final GitHubExtension gitHubExtension) {
+        gitHubApiClient.postUpdatedStatus(new GitHubStatus(gitHubStatusType), prSha)
+                .enqueue(new DefaultCallback<>());
+    }
 
+    @Nullable
+    public GitHubPRDetails getPRDetailsSync() {
         try {
-            Response<GitHubComment> gitHubCommentResponse = gitHubApiClient.postComment(new GitHubComment(comment), gitHubExtension.getIssueNumber()).execute();
-            return gitHubCommentResponse.isSuccessful() ? Status.OK : Status.FAIL;
-        } catch (IOException ignored) {
-            return Status.FAIL;
+            Response<GitHubPRDetails> gitHubPRDetailsResponse = gitHubApiClient.getPRDetails(gitHubExtension.getIssueNumber()).execute();
+            return gitHubPRDetailsResponse.body();
+        } catch (final Exception e) {
+            e.printStackTrace();
+            return null;
         }
     }
 
-    public Status postUpdatedGitHubStatus(GitHubStatusType gitHubStatusType, String sha) {
-
+    @NotNull
+    public List<Diff> getPRDiffsSync() {
         try {
-            Response<GitHubStatus> gitHubStatusResponse = gitHubApiClient.postUpdatedStatus(new GitHubStatus(gitHubStatusType), sha).execute();
-            return gitHubStatusResponse.isSuccessful() ? Status.OK : Status.FAIL;
-        } catch (IOException ignored) {
-            return Status.FAIL;
+            final Response<List<Diff>> gitHubPRDiffsResponse
+                    = gitHubApiClient.getPRDiffs(gitHubExtension.getIssueNumber()).execute();
+            
+            return gitHubPRDiffsResponse.body();
+        } catch (final Exception e) {
+            e.printStackTrace();
+            return new ArrayList<>();
         }
     }
 
-    public GitHubPullRequest getPullRequestDetails() {
+    public void postGitHubInlineCommentSync(
+            @NotNull final String body,
+            @NotNull final String prSha,
+            @NotNull final String relativeFilePath,
+            final int diffLineIndex) {
 
         try {
-            Response<GitHubPullRequest> gitHubPullRequestResponse = gitHubApiClient.getPullRequest(gitHubExtension.getIssueNumber()).execute();
-            return gitHubPullRequestResponse.body();
-        } catch (Exception ignored) {
-            ignored.printStackTrace();
-            return null;
+            gitHubApiClient.postInlineComment(
+                    new GitHubInlineComment(body, prSha, relativeFilePath, diffLineIndex), gitHubExtension.getIssueNumber())
+                            .execute();
+        } catch (final Exception e) {
+            e.printStackTrace();
+        }
+    }
+    
+    private static final class DefaultCallback<T> implements Callback<T> {
+
+        private DefaultCallback() {
+            // This constructor intentionally left blank.
+        }
+
+        @Override
+        public void onResponse(final Call<T> call, final Response<T> response) {
+            // This method intentionally left blank.            
```

Comment:
Could log more info here, but as I said above, was not receiving any responses to inline comment POSTs anyway...